com.unity.shadergraph/Editor/Drawing/Inspector/PropertyDrawers/ShaderInputPropertyDrawer.cs

```diff
@@ -196,6 +198,57 @@ void BuildExposedField(PropertySheet propertySheet)
             }
         }
 
+        void BuildCustomBindingField(PropertySheet propertySheet, AbstractShaderProperty property)
+        {
+            if (isSubGraph && property.isCustomSlotAllowed)
+            {
+                var toggleDataPropertyDrawer = new ToggleDataPropertyDrawer();
+                propertySheet.Add(toggleDataPropertyDrawer.CreateGUI(
+                    newValue =>
+                    {
+                        if (property.useCustomSlotLabel == newValue.isOn)
+                            return;
+                        this._preChangeValueCallback("Change Custom Binding");
+                        property.useCustomSlotLabel = newValue.isOn;
+                        graphData.ValidateGraph();
+                        this._postChangeValueCallback(true, ModificationScope.Topological);
+                    },
+                    new ToggleData(property.isConnectionTestable),
+                    "Use Custom Binding",
+                    out var exposedToggleVisualElement));
+                exposedToggleVisualElement.SetEnabled(true);
+
+                if (property.useCustomSlotLabel)
```

Comment:
Was a bit confused by the redundancy between this flag useCustomSlotLabel and the property isConnectionTestable when I looked under the hood and discovered they're the same thing, any reason why we couldn't just merge them both into a single flag?

Reply:
Oh, they do actually alias the same thing, it was just context for reading the code.